### PR TITLE
Payment Request button changes

### LINKED ISSUE #1563: Use cart totals for Payment Request Button in the product page
### Related issues for context:
- #1559
- #1459
- #641
- #950
- #1411
- #1064
- #1146
- #829
- #1113
- #1234
- #1579
- #1401

Because we allow users to start the payment request flow from the product page, the total price and `displayItems` for the Payment Request Button are calculated in three different ways in `WC_Stripe_Payment_Request`. Two of these ways are for the product page.

### This has some heavy downsides:

- Since WooCommerce doesn't provide a way to retrieve the product price through JavaScript, we need to manually calculate the price based on quantity, variations and other variables on the front-end.
- Before being added to the cart, each product type has its own traits to calculate the price (fees, taxes, recurring or one-time price, etc.), which are only normalized when added to the cart.
- Some extensions might include discounts, packages, or other types of filters which are applied only in the cart, making it impossible to checkout directly with the data we can gather from the product page.

### How to fix this:

**Essentially, the idea to fix this is to calculate totals and labels after the PRB is clicked, through some async request after the product is added to the cart.**

We can wait for the product to be added to the cart and then trigger another request for `ajax_get_cart_details` to always get the same price and labels, either from the product page or from the cart or checkout pages.

**Of course, this extra request is a UX tradeoff, which we need to find a way to address. Here are some ideas:**

1. Add loading indicators and block the PRB once it's clicked, then make the request for the prices, then open the PR dialog.
2. Find a way to add loading indicators to the native PR dialog instead of the button. For instance, Apple Pay already has a native loading state for the Apple Pay dialog right after you click the button, or before the payment is confirmed.

### Other ideas we could include in this fix:

- Since it's fairly easy to get the price for a simple product, and I believe that's the majority of the products, we could perhaps bypass this extra request for certain types of products. Keeping it the way it is for the products we're 100% able to calculate the price without triggering the cart endpoint - I still have some concerns with this approach, because although we can calculate the price for that product type, other extensions which add filters to the cart might still make it not compatible.
- If the PR flow was initiated from the product page, but later canceled by the user, it might be desirable to remove that item from the cart.
- Ideally, we should also consider normalizing label itemization for this fix. Either have the PR dialog in the cart and checkout pages display each item in the cart, or have the product page not display each item. I personally find it useful to display the quantity of an item (e.g. Album (x2)), but currently that only applies to the product page.

### PR DESCRIPTION
**Pull request by asumaran**
Fixes #1563

#### Product types

- [x] simple #1577
- [x] variable #1587
- [x] variation #1587
- [ ] ~subscription~, ~variable-subscription~, ~subscription_variation~ Handled in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1609
- [ ] ~booking~ Will create a separate issue. Requires extra work in both sides (ours and the plugin) to make it work.
- [x] Various improvements #1597
- [ ] bundle #1607
- [ ] composite #1608
- [ ] mix-and-match #1610

### CODE REVIEW

**Comment by asumaran** [comment]
@ricardo this PR contains approved changes only (#1577 and #1587). I was planning to merge this PR so everyone can test and catch any issue that may come up.

For the record #1563 is still not complete since #1624 is required to complete support for Bundles and Composite products. I'm planning to work on it and complete #1563 for remaining product types.

If this is PR merged [Mix and Match support](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1610) can be merged directly to trunk.

What do you think? I've tested this PR and all existing functionality is preserved and even improved in some cases.

File: includes/payment-methods/class-wc-stripe-payment-request.php [hunk 1]
```diff
@@ -723,7 +667,7 @@ public function javascript_params() {
 	 * Load public scripts and styles.
 	 *
 	 * @since   3.1.0
-	 * @version 5.2.0
+	 * @version x.x.x
```
Comment (ricardo) [inline includes/payment-methods/class-wc-stripe-payment-request.php]: `function scripts` is not getting changed now, so we can revert `x.x.x`.

**Review by ricardo — APPROVED** [review]
As I had already tested and reviewed the previous PRs, I smoke tested the main flows with different types of products again - Simple, Variable, Subscriptions, and took another look at the code.

LGTM 🚢 

> What do you think? I've tested this PR and all existing functionality is preserved and even improved in some cases.

I think it's fine to merge it and add support for the remaining products with #1624.

For now it would be interesting to create detailed manual testing steps for GlobalStep if they do start testing Stripe as well.

Good job on these PRs 🙌.

Before you merge, should we add a changelog entry?

File: includes/payment-methods/class-wc-stripe-payment-request.php (hunk 1 above)
Comment (asumaran) [reply includes/payment-methods/class-wc-stripe-payment-request.php]: I'll update and add also a changelog entry

**Pull request merged**